src/db_process.c

```diff
@@ -267,9 +267,33 @@ static int vm_info_query_callback(void *all_vm_info_ptr, int argc, char **argv,
         vm_info_captured.start_addr = strdup(argv[0]);
         vm_info_captured.end_addr = strdup(argv[1]);
         vm_info_captured.mmap_path = strdup(argv[2]);
-     	vm_info_captured.kve_protection = atoi(argv[3]);
-     	vm_info_captured.mmap_flags = atoi(argv[4]); 
-        vm_info_captured.vnode_type = atoi(argv[5]);
+
+	// Checking that the received values are valid integers
+	// as expected from the vm_info_captured data structure. 
+	// If the check fails, it means the data is invalid. Without
+	// a more sophisticated logic in place to handle invalid data, 
+	// we will just quit the program and let users to examine 
+	// and fix the data before running this function again.
+	char *prot_pEnd;
+	long int prot = strtol(argv[3], &prot_pEnd, 10);
+	if (*prot_pEnd != '\0') {
+		errx(1, "Invalid kve_protection, it should be an integer: %s", argv[3]);
+	}
```

Comment:
I'd be tempted to roll this into a utility function to avoid this single use variables.

Reply:
Addressed